geopandas/geodataframe.py

```diff
@@ -252,9 +252,20 @@ def set_geometry(self, col, drop=False, inplace=False, crs=None):
 
         Parameters
         ----------
-        col : column label or array
+        col : column label or array-like
+            An existing column name or values to set as the new geometry column.
+            If values (array-like, (Geo)Series) are passed, then if they are named
+              (Series) the new geometry column will have the corresponding name,
+              otherwise the existing geometry column will be replaced. If there is
+              no existing geometry column, the new geometry column will use the
+              default name "geometry".
```

Comment:
```suggestion
            If values (array-like, (Geo)Series) are passed, then if they are named
            (Series) the new geometry column will have the corresponding name,
            otherwise the existing geometry column will be replaced. If there is
            no existing geometry column, the new geometry column will use the
            default name "geometry".
```